spec/link/group-group/schema.js

```diff
@@ -0,0 +1,60 @@
+const { feedId, messageId, cloakedMessageId, tangle } = require('ssb-schema-definitions')()
+
+const schema = {
+  $schema: 'http://json-schema.org/schema#',
+  type: 'object',
+  required: ['type', 'parent', 'child', 'recps', 'tangles'],
+  properties: {
+    type: {
+      type: 'string',
+      pattern: '^link/group-group'
+    },
+    parent: { $ref: '#/definitions/cloakedMessageId' },
+    child: { $ref: '#/definitions/cloakedMessageId' },
+
+    kaitiaki: {
+      type: 'object',
+      required: ['set'],
+      properties: {
+        set: { type: 'boolean' }
+      },
+      additionalProperties: false
+    },
+    subgroup: {
+      type: 'object',
+      required: ['set'],
+      properties: {
+        set: { type: 'boolean' }
+      },
+      additionalProperties: false
+    },
+
+    recps: {
+      type: 'array',
+      items: [
+        { $ref: '#/definitions/cloakedMessageId' }
+      ],
+      minItems: 1,
+      maxItems: 1
+    },
+
+    tangles: {
+      type: 'object',
+      required: ['link'],
+      properties: {
+        link: { $ref: '#/definitions/tangle/root' }
+      }
+    }
+  },
+  additionalProperties: false,
+  definitions: {
+    feedId,
+    messageId,
+    cloakedMessageId,
+    tangle: { root: tangle.root }
+  }
+}
+
+module.exports = {
+  root: JSON.stringify(schema, null, 2)
+}
```

Comment:
yeah this is painful to look at now!
I think the only thing it looks like we need to do is add a pre-create check that there is only one recps and it's a cloakedMessageId

Reply:
Hahaha was hoping you wouldnt notice but i knew you would :P All good i'll do a refactor :)